pkg/config/manifest.go

```diff
@@ -566,220 +568,73 @@ func (s *Step) GetMixinName() string {
 	return mixinName
 }
 
-// ResolveStep will walk through the Step's data and resolve any placeholder
-// data using the definitions in the manifest, like parameters or credentials.
-func (m *Manifest) ResolveStep(step *Step) error {
-	return reflectwalk.Walk(step, m)
-}
+func (m *Manifest) buildSourceData() (map[string]interface{}, error) {
+	data := make(map[string]interface{})
+	m.sensitiveValues = []string{}
+	bundle := make(map[string]interface{})
+	data["bundle"] = bundle
```

Comment:
This is a question about mustache. Looks like we are building up a map that has keys for parameters, outputs, etc inside of it, then that's what the template is operating upon.

If we had a struct called "TemplateMagicPants" (or whatever) and it had fields for parameters, outputs, credentials, and everything else we needed, would that work too as input data to mustache? Or would that result in different syntax in the template?

This is just a question for me to understand mustache, not asking you to change anything. 😀

Reply:
We'd need to have a struct called TemplateMagicPants that had those, that lived in a map with the key bundle. Or another struct. It's walking what ever the template syntax is. I was trying to keep it _mostly_ the same, and I think the comment from our Amazon friends used similar syntax.

I did find myself wondering what the point of the "bundle" is. everything is rooted off that....soooooo maybe just {{ parameters.x }} or {{ outputs.y }} or {{ credentials.z }} or {{ dependencies.x.y.z }} 

I was thinking along these lines for the general refactor of "the manifest" from the running state of things, but here I didn't want to change the syntax of the template since we hadn't chatted about that. Happy to go either way, but I think my opinion is the "bundle" key is not really needed?

Reply:
Let's not remove the `bundle.` prefix for now. I believe that it is important for both giving people a starting point in their heads for what to type, and also for the autocomplete to have something to go off of.